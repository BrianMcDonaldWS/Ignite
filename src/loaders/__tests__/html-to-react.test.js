import {
  getLink,
  addActive,
  addActiveAll,
  index,
  replaceAt,
  insertBreaks,
  regexIndexOf,
  replaceIdLinks,
  sanitizeJSX,
  markDownPage,
  detectIndex
} from '../html-to-react';

test('getLink', () => {
  expect(getLink('<a href="/foo">Foo</a>')).toBe('foo');
  expect(getLink('<a>Foo</a>')).toBe('');
});

test('replaceAt', () => {
  expect(replaceAt('A dog barked loudly', 'dog', 'wolf', 2)).toBe(
    'A wolf barked loudly'
  );
});

test('addActive', () => {
  expect(
    addActive('<a href="/foo">Foo</a>', 'foo', 'bar', 'index.md')
  ).toMatchSnapshot();
});

test('addActiveAll', () => {
  expect(
    addActiveAll(
      `
      <a href="/foo">Foo</a>'
      <a href="/bar">Foo</a>'
    `,
      'bar',
      'index.md'
    )
  ).toMatchSnapshot();
});

test('index', () => {
  expect(index('<ul></ul>', '/path', {})).toMatchSnapshot();
  expect(index('<p></p>', '/path', {})).toMatchSnapshot();
});

test('insertBreaks', () => {
  expect(
    insertBreaks(`
    <div>
      <pre>
        Line
        with
        breaks
      </pre>

      <pre>
        Another
        Line
        with
        breaks
      </pre>
    </div>
  `)
  ).toMatchSnapshot();
});

test('regexIndexOf', () => {
  expect(regexIndexOf('I am the test:string.', /[\S]+:[\S]+/)).toBe(9);
  expect(regexIndexOf('I am the test:string.', /[\S]+:[\S]+/, 15)).toBe(-1);
});

test('replaceIdLinks', () => {
  expect(
    replaceIdLinks(`
    <a href="#some-id">Link</a>
    <a className="fas fa-hashtag headerLink" href="#some-other-id">Link</a>
  `)
  ).toMatchSnapshot();
});

test('sanitizeJSX', () => {
  expect(
    sanitizeJSX('<a className="/foo">{Foo}</a>', '/path', {})
  ).toMatchSnapshot();
  expect(
    sanitizeJSX('<a className=!{"/foo"!}>Foo</a>', '/path', {})
  ).toMatchSnapshot();

  expect(sanitizeJSX('<br>', '/path', {})).toMatchSnapshot();
  expect(sanitizeJSX('<pluginprovider>', '/path', {})).toMatchSnapshot();
  expect(
    sanitizeJSX('<div class="checkbox" />', '/path', {})
  ).toMatchSnapshot();
  expect(
    sanitizeJSX('<label for="checkbox" />', '/path', {})
  ).toMatchSnapshot();
  expect(
    sanitizeJSX('<input type="checkbox" />', '/path', {})
  ).toMatchSnapshot();
  expect(
    sanitizeJSX(
      `
      <pre>
        content
      </pre>
    `,
      '/path',
      {}
    )
  ).toMatchSnapshot();
});

test('markDownPage', () => {
  expect(
    markDownPage(`<span class="highlighted-line"></span>`)
  ).toMatchSnapshot();
});

test('detectIndex', () => {
  const options = {
    index: 'index.md'
  };

  expect(detectIndex('path/to/file.md', 'file.md', options)).toBe(false);
  expect(detectIndex('path/to/index.md', 'index.md', options)).toBe(true);

  options.navItems = {
    root: '/',
    pages: 'pages'
  };

  expect(detectIndex('path/to/index.md', 'index.md', options)).toBe(true);
  expect(detectIndex('path/to/pages/index.md', 'pages/index.md', options)).toBe(
    true
  );
});
