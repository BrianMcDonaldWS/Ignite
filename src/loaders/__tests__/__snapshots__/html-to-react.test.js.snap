// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addActive 1`] = `
"<a className=!{
        '/foo' === props.currentPage ||
        ('bar' === 'foo' && ('/' === props.currentPage || '/index.html' === props.currentPage)) ||
        ('bar' === 'foo' && props.currentPage && props.currentPage.includes('index.md'))
          ? 'is-active'
          : null
      !}
      href=\\"/foo\\">Foo</a>"
`;

exports[`addActiveAll 1`] = `
"
      <a className=!{
        '/' === props.currentPage ||
        ('bar' === '' && ('/' === props.currentPage || '/index.html' === props.currentPage)) ||
        ('bar' === '' && props.currentPage && props.currentPage.includes('index.md'))
          ? 'is-active'
          : null
      !}
      href=\\"\\"></a>'
      <a className=!{
        '/foo' === props.currentPage ||
        ('bar' === 'foo' && ('/' === props.currentPage || '/index.html' === props.currentPage)) ||
        ('bar' === 'foo' && props.currentPage && props.currentPage.includes('index.md'))
          ? 'is-active'
          : null
      !}
      href=\\"/foo\\">Foo</a>'
      <a className=!{
        '/bar' === props.currentPage ||
        ('bar' === 'bar' && ('/' === props.currentPage || '/index.html' === props.currentPage)) ||
        ('bar' === 'bar' && props.currentPage && props.currentPage.includes('index.md'))
          ? 'is-active'
          : null
      !}
      href=\\"/bar\\">Foo</a>'
    "
`;

exports[`index 1`] = `
"
    
      import React from 'react';
      import makeClass from 'classnames';
      import { Link } from 'react-router-dom';
      import Gist from 'react-gist';
      import TweetEmbed from 'react-tweet-embed'

      const OptionalLink = props => props.to
        ? <Link {...props} />
        : <a { ...props} />;

      const PluginProvider = ({plugins, name, options, children}) => {
        let Plugin = plugins[name];
        const pluginOptions = Plugin.options;

        if (!Plugin) {
          return <div />;
        }

        Plugin = Plugin.component;
        return <Plugin {...pluginOptions}  children={children} {...options} />;
      };

      
    

    export default function markDownPage(props) {
      return (
        <aside className={makeClass('menu', props.className)} onClick={props.onClick}>
          <ul className=\\"menu-list\\"></ul>
        </aside>
      );
    }

    window.configuration.setFirstLink('/path', '/');
  "
`;

exports[`index 2`] = `
"
    
      import React from 'react';
      import makeClass from 'classnames';
      import { Link } from 'react-router-dom';
      import Gist from 'react-gist';
      import TweetEmbed from 'react-tweet-embed'

      const OptionalLink = props => props.to
        ? <Link {...props} />
        : <a { ...props} />;

      const PluginProvider = ({plugins, name, options, children}) => {
        let Plugin = plugins[name];
        const pluginOptions = Plugin.options;

        if (!Plugin) {
          return <div />;
        }

        Plugin = Plugin.component;
        return <Plugin {...pluginOptions}  children={children} {...options} />;
      };

      
    

    export default function markDownPage(props) {
      return (
        <aside className={makeClass('menu', props.className)} onClick={props.onClick}>
          <p className=\\"menu-label\\"></p>
        </aside>
      );
    }

    window.configuration.setFirstLink('/path', '/');
  "
`;

exports[`insertBreaks 1`] = `
"
    <div>
      <pre><br />        Line<br />        with<br />        breaks<br />      </pre>

      <pre><br />        Another<br />        Line<br />        with<br />        breaks<br />      </pre>
    </div>
  "
`;

exports[`markDownPage 1`] = `
"
    
      import React from 'react';
      import makeClass from 'classnames';
      import { Link } from 'react-router-dom';
      import Gist from 'react-gist';
      import TweetEmbed from 'react-tweet-embed'

      const OptionalLink = props => props.to
        ? <Link {...props} />
        : <a { ...props} />;

      const PluginProvider = ({plugins, name, options, children}) => {
        let Plugin = plugins[name];
        const pluginOptions = Plugin.options;

        if (!Plugin) {
          return <div />;
        }

        Plugin = Plugin.component;
        return <Plugin {...pluginOptions}  children={children} {...options} />;
      };

      
    

    const markDownPage = props => (
      <div className={props.className}>
        <section>
          <span className=\\"highlighted-line hero is-primary\\"></span>
        </section>
      </div>
    );

    export default markDownPage;
  "
`;

exports[`replaceIdLinks 1`] = `
"
    <OptionalLink currentPage={(this && this.props || props).currentPage} to=\\"#some-id\\">Link</OptionalLink>
    <OptionalLink currentPage={(this && this.props || props).currentPage} className=\\"fas fa-hashtag headerLink\\" to=\\"#some-other-id\\">Link</OptionalLink>
  "
`;

exports[`sanitizeJSX 1`] = `"<OptionalLink currentPage={(this && this.props || props).currentPage} className=\\"/foo\\">&#123;Foo&#125;</OptionalLink>"`;

exports[`sanitizeJSX 2`] = `"<OptionalLink currentPage={(this && this.props || props).currentPage} className={\\"/foo\\"}>Foo</OptionalLink>"`;

exports[`sanitizeJSX 3`] = `"<br/>"`;

exports[`sanitizeJSX 4`] = `"<PluginProvider>"`;

exports[`sanitizeJSX 5`] = `"<div className=\\"field\\" />"`;

exports[`sanitizeJSX 6`] = `"<label htmlFor=\\"field\\" />"`;

exports[`sanitizeJSX 7`] = `"<input type=\\"checkbox\\" readOnly className=\\"is-checkradio\\" />"`;

exports[`sanitizeJSX 8`] = `
"
      <pre><br />        content<br />      </pre>
    "
`;

exports[`sanitizeJSX- complex 1`] = `
"
    <OptionalLink currentPage={(this && this.props || props).currentPage} className=\\"fas fa-hashtag headerLink\\" to=\\"#zero-hassle-documentation\\" aria-hidden=\\"true\\" />
    <OptionalLink currentPage={(this && this.props || props).currentPage} to=\\"#/pages/GettingStarted.md\\">&#x1F389; Get Started</OptionalLink>
  "
`;
